### ISSUE TITLE
Make binary resolving more flexible

### ISSUE THREAD

**Issue by tokee** [netarchivesuite/solrwayback#356]
The current abstraction of resource (WARC records) resolving expects `[WARC-filename, offset]`. By extending this to `[WARC-filename, offset, timestamp, URL]` it should be possible to use PyWB as backend for resolving. This would work well for setups that already have PyWB running and don't want to add a new exposure point for their WARCs.

**Comment by thomasegense** [comment]
Basically implement the CDX-index API. 

Notice this would make #388 obsolete.

**Comment by tokee** [comment]
No, quite the reverse: This allows SolrWayback to use an existing CDX-index API for resource retrieval.
